packages/calling-stateful-client/src/CallAgentDeclarativeCommon.ts

```diff
@@ -0,0 +1,287 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { Call, CallAgent, CallEndReason, CollectionUpdatedEvent, IncomingCall } from '@azure/communication-calling';
+/* @conditional-compile-remove(teams-call) */
+import { TeamsCall, TeamsCallAgent, TeamsIncomingCall } from '@azure/communication-calling';
+import { CallCommon } from './CallCommonDeclarative';
+import { CallContext } from './CallContext';
+import { callDeclaratify, DeclarativeCall } from './CallDeclarative';
+import { CallSubscriber } from './CallSubscriber';
+import {
+  convertSdkCallToDeclarativeCall,
+  convertSdkIncomingCallToDeclarativeIncomingCall,
+  isACSCall,
+  isACSCallAgent
+} from './Converter';
+import { DeclarativeIncomingCall, incomingCallDeclaratify } from './IncomingCallDeclarative';
+import { IncomingCallSubscriber } from './IncomingCallSubscriber';
+import { InternalCallContext } from './InternalCallContext';
+import { disposeAllViewsFromCall } from './StreamUtils';
+/* @conditional-compile-remove(teams-call) */
+import { DeclarativeTeamsCall, teamsCallDeclaratify } from './TeamsCallDeclarative';
```

Comment:
Remove them as much as we can, still one teamsCallAgent is needed for "get" function because we need some type guard there for function names and parameters